src/createServer.js

```diff
@@ -1,8 +1,89 @@
+/* eslint-disable no-console */
 'use strict';
 
+const http = require('node:http');
+const { formidable } = require('formidable');
+const fs = require('node:fs');
+const path = require('node:path');
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = new http.Server();
+
+  server.on('request', async (req, res) => {
+    if (req.url === '/' || req.url === '/favicon.ico') {
+      res.on('error', (err) => {
+        console.error(err);
+        res.statusCode = 400;
```

Comment:
The status code here should be 500 or another appropriate error code, as 400 is typically used for client errors, not server errors.